### PR TITLE
fix:[Fluid Devtools] npm run test:coverage for browser extension OOMs locally and makes CI builds noticeably longer

### PR DESCRIPTION
**Pull request by chentong7**
## Description

`npm run test:coverage` time duration bump from ~1m20s to ~3m30s. Example pipeline run: https://dev.azure.com/fluidframework/internal/_build/results?buildId=134923&view=logs&j=3dc8fd7e-4368-5a92-293e-d53cefc8c4b3&t=38623822-d55d-5839-9078-a2164c1dcf41
 Example error:
~~~
2023-03-02T02:15:42: File '/mnt/vss/_work/1/s/packages/tools/client-debugger/client-debugger-chrome-extension/dist/background/@***-tools/client-debugger-chrome-extension/src/background/BackgroundScript.ts' does not exist (any more).
2023-03-02T02:15:42: File '/mnt/vss/_work/1/s/packages/tools/client-debugger/client-debugger-chrome-extension/dist/background/@***-tools/client-debugger-chrome-extension/src/background/Logging.ts' does not exist (any more).
2023-03-02T02:15:42: File '/mnt/vss/_work/1/s/packages/tools/client-debugger/client-debugger-chrome-extension/dist/background/@***-tools/client-debugger-chrome-extension/src/messaging/Constants.ts' does not exist (any more).
2023-03-02T02:15:42: File '/mnt/vss/_work/1/s/packages/tools/client-debugger/client-debugger-chrome-extension/dist/background/@***-tools/client-debugger-chrome-extension/src/messaging/index.ts' does not exist (any more).
2023-03-02T02:15:42: File '/mnt/vss/_work/1/s/packages/tools/client-debugger/client-debugger-chrome-extension/dist/background/@***-tools/client-debugger-chrome-extension/src/messaging/Messages.ts' does not exist (any more).
2023-03-02T02:15:42: File '/mnt/vss/_work/1/s/packages/tools/client-debugger/client-debugger-chrome-extension/dist/background/@***-tools/client-debugger-chrome-extension/src/messaging/Utilities.ts' does not exist (any more).
2023-03-02T02:15:42: File '/mnt/vss/_work/1/s/packages/tools/client-debugger/client-debugger-chrome-extension/dist/background/@***-tools/client-debugger-chrome-extension/webpack/bootstrap' does not exist (any more).
2023-03-02T02:15:42: File '/mnt/vss/_work/1/s/packages/tools/client-debugger/client-debugger-chrome-extension/dist/background/@***-tools/client-debugger-chrome-extension/webpack/runtime/compat get default export' does not exist (any more).
2023-03-02T02:15:42: File '/mnt/vss/_work/1/s/packages/tools/client-debugger/client-debugger-chrome-extension/dist/background/@***-tools/client-debugger-chrome-extension/webpack/runtime/define property getters' does not exist (any more).
2023-03-02T02:15:42: File '/mnt/vss/_work/1/s/packages/tools/client-debugger/client-debugger-chrome-extension/dist/background/@***-tools/client-debugger-chrome-extension/webpack/runtime/ensure chunk' does not exist (any more).
2023-03-02T02:15:42: File '/mnt/vss/_work/1/s/packages/tools/client-debugger/client-debugger-chrome-extension/dist/background/@***-tools/client-debugger-chrome-extension/webpack/runtime/get javascript chunk filename' does not exist (any more).
2023-03-02T02:15:42: File '/mnt/vss/_work/1/s/packages/tools/client-debugger/client-debugger-chrome-extension/dist/background/@***-tools/client-debugger-chrome-extension/webpack/runtime/global' does not exist (any more).
2023-03-02T02:15:42: File '/mnt/vss/_work/1/s/packages/tools/client-debugger/client-debugger-chrome-extension/dist/background/@***-tools/client-debugger-chrome-extension/webpack/runtime/hasOwnProperty shorthand' does not exist (any more).
2023-03-02T02:15:42: File '/mnt/vss/_work/1/s/packages/tools/client-debugger/client-debugger-chrome-extension/dist/background/@***-tools/client-debugger-chrome-extension/webpack/runtime/jsonp chunk loading' does not exist (any more).
2023-03-02T02:15:42: File '/mnt/vss/_work/1/s/packages/tools/client-debugger/client-debugger-chrome-extension/dist/background/@***-tools/client-debugger-chrome-extension/webpack/runtime/load script' does not exist (any more).
2023-03-02T02:15:42: File '/mnt/vss/_work/1/s/packages/tools/client-debugger/client-debugger-chrome-extension/dist/background/@***-tools/client-debugger-chrome-extension/webpack/runtime/make namespace object' does not exist (any more).
2023-03-02T02:15:42: File '/mnt/vss/_work/1/s/packages/tools/client-debugger/client-debugger-chrome-extension/dist/background/@***-tools/client-debugger-chrome-extension/webpack/runtime/node module decorator' does not exist (any more).
2023-03-02T02:15:42: File '/mnt/vss/_work/1/s/packages/tools/client-debugger/client-debugger-chrome-extension/dist/background/@***-tools/client-debugger-chrome-extension/webpack/runtime/publicPath' does not exist (any more).
2023-03-02T02:15:42: File '/mnt/vss/_work/1/s/packages/tools/client-debugger/client-debugger-chrome-extension/dist/background/@***-tools/client-debugger/dist/AudienceMetadata.js' does not exist (any more).
2023-03-02T02:15:42: File '/mnt/vss/_work/1/s/packages/tools/client-debugger/client-debugger-chrome-extension/dist/background/@***-tools/client-debugger/dist/Container.js' does not exist (any more).
2023-03-02T02:15:42: File '/mnt/vss/_work/1/s/packages/tools/client-debugger/client-debugger-chrome-extension/dist/background/@***-tools/client-debugger/dist/FluidClientDebugger.js' does not exist (any more).
2023-03-02T02:15:42: File '/mnt/vss/_work/1/s/packages/tools/client-debugger/client-debugger-chrome-extension/dist/background/@***-tools/client-debugger/dist/index.js' does not exist (any more).
2023-03-02T02:15:42: File '/mnt/vss/_work/1/s/packages/tools/client-debugger/client-debugger-chrome-extension/dist/background/@***-tools/client-debugger/dist/Registry.js' does not exist (any more).
2023-03-02T02:15:42: File '/mnt/vss/_work/1/s/packages/tools/client-debugger/client-debugger-chrome-extension/dist/background/@***-tools/client-debugger/dist/messaging/Constants.js' does not exist (any more).
2023-03-02T02:15:42: File '/mnt/vss/_work/1/s/packages/tools/client-debugger/client-debugger-chrome-extension/dist/background/@***-tools/client-debugger/dist/messaging/index.js' does not exist (any more).

~~~
Remove unnecessary file from test coverage path.

### CODE REVIEW

**Comment by Josmithr** [comment]
@chentong7 I'm not totally sure what this PR is trying to do. I think the coverage config in browser-extension's package.json was wrong, but I don't think we want to remove dist from the configs altogether. We probably just want to point it to `dist/tsc` instead of just `dist`. And I don't think we want to make any changes to the config in devtools-core.

**Comment by Josmithr** [comment]
Nit: The PR title's use of the Conventional Commits format is a bit off. I think you want `fix(devtools): ...`

**Comment by alexvy86** [comment]
> We probably just want to point it to dist/tsc instead of just dist (since dist/tsc is where our non-webpack build output is generated).

Agree with Josh here. The `dist/` folder for the browser extension has several subfolders with output from different "pieces" of the build process, and the only one we should check is the output of the TypeScript compilation for the code from the package itself that is actually published in the end. I don't think the one for e2e-tests is necessary to include in code coverage, and definitely not the webpack bundle.

image

> For the error output in the PR description - those errors appear to be referencing the old package names and the files in them

Yeah, this sounds to me like a local run of the command, on an environment that still has old build output from when those packages existed. I imagine it probably gets fixed with `npm run clean; npm run build` for the package before running the code coverage report.

File: packages/tools/devtools/devtools-browser-extension/package.json [hunk 1]
```diff
@@ -53,8 +53,7 @@
 		],
 		"exclude-after-remap": false,
 		"include": [
-			"src/**/*.ts",
-			"dist/**/*.js"
+			"src/**/*.ts"
```
Comment (alexvy86) [inline packages/tools/devtools/devtools-browser-extension/package.json:56]: @chentong7 can you try running a pipeline with this here? (And you can revert the change to `devtools-core`. I tested locally and this accomplishes what I think we need to fix this. Basically telling c8 that the only files within `dist` that it should be checking for coverage are the ones that resulted from compiling the ones in `src`. The other folders under `dist` are for other purposes, and I'm not sure exactly why, but if we let c8 look at all of them as potential source files to be checked for coverage, it ends up generating a _ton_ of extra output under `nyc/report`, including a `node_modules` folder, which explains the huge amount of files that it's looking for and failing to find when the pipeline tries to publish the code coverage report.
```suggestion
			"src/**/*.ts"
			"dist/tsc/**/*.js"			
```

If you run `npx c8 --clean npm run test:jest` on this package locally, with and without this change, you'll see in the output how without this change the list of files is huge, but with this change the list of files is much shorter, correctly limited to only the source files for the package.

Comment (chentong7) [reply packages/tools/devtools/devtools-browser-extension/package.json:56]: Seems like this is working. Tested in remote branch: https://github.com/microsoft/FluidFramework/pull/17471

**Pull request closed without merging**